src/if-run/lib/regroup.ts

```diff
@@ -0,0 +1,87 @@
+import {z} from 'zod';
+import {ERRORS} from '@grnsft/if-core/utils';
+import {PluginParams} from '@grnsft/if-core/types';
+
+import {validate} from '../../common/util/validations';
+
+import {STRINGS} from '../config';
+
+const {InvalidGroupingError} = ERRORS;
+
+const {INVALID_GROUP_KEY, REGROUP_ERROR} = STRINGS;
+
+/**
+ * Grouping strategy.
+ */
+export const Regroup = (inputs: PluginParams[], groups: string[]) => {
+  /**
+   * Creates structure to insert inputs by groups.
+   */
+  const appendGroup = (
```

Comment:
create some options property with custom type, or at least define the `object` property type and remove the `any` return type